source/extensions/common/wasm/wasm.cc

```diff
@@ -1279,6 +1342,136 @@ uint32_t Context::getResponseCode(StreamType type) {
   return streamInfo->responseCode().value_or(500);
 }
 
+template <int N> struct marshall { char buf[N]; };
+
+absl::optional<std::string> serializeValue(Filters::Common::Expr::CelValue value) {
+  using Filters::Common::Expr::CelValue;
+  switch (value.type()) {
+  case CelValue::Type::kMessage:
+    if (value.MessageOrDie() != nullptr) {
+      std::string result;
+      value.MessageOrDie()->SerializeToString(&result);
+      return result;
+    }
+  case CelValue::Type::kString:
+    return std::string(value.StringOrDie().value());
+  case CelValue::Type::kBytes:
+    return std::string(value.BytesOrDie().value());
+  case CelValue::Type::kInt64: {
+    const size_t n = sizeof(int64_t);
+    marshall<n> out = absl::bit_cast<marshall<n>>(value.Int64OrDie());
+    return std::string(out.buf, n);
+  }
+  case CelValue::Type::kUint64: {
+    const size_t n = sizeof(uint64_t);
+    marshall<n> out = absl::bit_cast<marshall<n>>(value.Uint64OrDie());
+    return std::string(out.buf, n);
+  }
+  case CelValue::Type::kDouble: {
+    const size_t n = sizeof(double);
+    marshall<n> out = absl::bit_cast<marshall<n>>(value.DoubleOrDie());
+    return std::string(out.buf, n);
+  }
+  case CelValue::Type::kBool: {
+    const size_t n = sizeof(bool);
+    marshall<n> out = absl::bit_cast<marshall<n>>(value.BoolOrDie());
+    return std::string(out.buf, n);
+  }
+  case CelValue::Type::kDuration: {
+    const size_t n = sizeof(absl::Duration);
+    marshall<n> out = absl::bit_cast<marshall<n>>(value.DurationOrDie());
+    return std::string(out.buf, n);
+  }
+  case CelValue::Type::kTimestamp: {
+    const size_t n = sizeof(absl::Time);
+    marshall<n> out = absl::bit_cast<marshall<n>>(value.TimestampOrDie());
+    return std::string(out.buf, n);
+  }
+  default:
+    // TODO: lists and maps
+    break;
+  }
+
+  return {};
+}
+
+absl::optional<std::string> Context::resolve(std::initializer_list<absl::string_view> parts) {
```

Comment:
Can you add some doc on resolve ?
The tests would be authoritative, but doc will help here.